samples/news/News.UITests/Pages/FavoritesPage.cs

```diff
@@ -12,17 +13,43 @@ public class FavoritesPage : BasePage
         };
 
         protected readonly Query FavoriteButton;
+        protected readonly Query EmptyLabel;
+        protected readonly Query NewsInfiniteListView;
 
         public FavoritesPage()
         {
             FavoriteButton = x => x.Marked(nameof(FavoriteButton));
+            NewsInfiniteListView = x => x.Marked(nameof(NewsInfiniteListView));
+            EmptyLabel = x => x.Marked(nameof(EmptyLabel));
         }
 
-        public void RemoveFavorite()
+        public FavoritesPage VerifyThatEmpty()
+        {
+            app.WaitForElement(EmptyLabel);
+            app.Screenshot("Verified that favorite is empty");
+            return this;
+        }
+
+        public FavoritesPage VerifyThatNotEmpty()
+        {
+            app.WaitForNoElement(EmptyLabel);
```

Comment:
nit: since this is a verification, I would swap `WaitForNoElement` with an `Assert.IsNull` or `Assert.IsEmpty`, mostly because it's a state verification.

Reply:
@Sweekriti91 could you please clarify this, at the moment of the verification element could be still there (animation, refresh, etc), so we specifically ask the framework to wait while the element disappears. And if it doesn't happen for the time allotted, the test fails. Otherwise, we know that element is no longer there and there is no need to do an extra assert. 

Did I properly understand your suggestion?

Reply:
Aha I see what you're doing. Leave this as a WaitFor then, it works for this test.